chain/network/src/network_protocol/network.proto

```diff
@@ -0,0 +1,296 @@
+syntax = "proto3";
+package network;
+
+// Wrapper of borsh-encoded PublicKey.
+// https://github.com/near/nearcore/blob/1a4edefd0116f7d1e222bc96569367a02fe64199/core/crypto/src/signature.rs#L201
+message PublicKey {
+  bytes borsh = 1;
```

Comment:
What's the plan for these in the future? Are we going to continue using `borsh` encoding for public keys indefinitely or would we consider eventually making it pure protobuf thing?

If the plan is to eventually change the representation here, `message` is probably not an appropriate type here unless compatible encoding is not a concern.

Reply:
All the message types storing currently a borsh blob are available for migration, by just adding regular message fields and deprecating "borsh" field. It is an explicit goal to make further migration borsh -> protobuf for these message types feasible and easy.

Reply:
> Adding message fields in a compatible manner requires to make the new ones optional, though

That's the neat part. In proto3 all fields are optional :). And it makes sense: otherwise you would have a bunch of special "original" fields and everything else would be "optional because added later". Having non-optional fields (which is a feature of proto2) makes it impossible to truly deprecate old fields.

> I feel like oneof PublicKey { BorshPublicKey borsh = 1; } could be more appropriate. 

Putting a field into a new oneof is a backward compatible change. No need to do it in advance.